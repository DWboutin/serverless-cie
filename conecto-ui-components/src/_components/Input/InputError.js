import React from 'react'
import PropTypes from 'prop-types'

const InputError = ({ children }) => (
  <div className="to-input-error">{children}</div>
)

InputError.defaultProps = {}

InputError.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    .isRequired,
}

export default InputError
