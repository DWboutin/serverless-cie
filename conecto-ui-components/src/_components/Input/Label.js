import React from 'react'
import PropTypes from 'prop-types'

const Label = ({ htmlFor, className, id, children }) => (
  <label htmlFor={htmlFor} className={className} id={id}>
    {children}
  </label>
)

Label.defaultProps = {}

Label.propTypes = {
  htmlFor: PropTypes.string,
  className: PropTypes.string,
  id: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    .isRequired,
}

export default Label
