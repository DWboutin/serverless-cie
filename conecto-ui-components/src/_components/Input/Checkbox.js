import React from 'react'
import PropTypes from 'prop-types'
import omit from 'omit.js'
import classNames from 'classnames'

import InputError from './InputError'

class Checkbox extends React.PureComponent {
  constructor(props) {
    super(props)

    this.state = {
      value: props.value || props.checked || false,
      focus: props.focused || false,
    }

    this.checkbox = React.createRef()

    this.classesPrefix = 'to-checkbox'

    this.handleKeyDown = this.handleKeyDown.bind(this)
    this.handleFieldChange = this.handleFieldChange.bind(this)
    this.handleFieldBlur = this.handleFieldBlur.bind(this)
    this.handleFieldFocus = this.handleFieldFocus.bind(this)
  }

  static getDerivedStateFromProps(nextProps) {
    if (nextProps && nextProps.value) {
      return {
        value: nextProps.value,
      }
    }
    return null
  }

  handleKeyDown(e) {
    const { disabled } = this.props

    if (!disabled && e.keyCode === 13) {
      this.setState(
        {
          value: !this.state.value,
        },
        () => {
          this.checkbox.current.dispatchEvent(
            new MouseEvent('click', {
              view: window,
              bubbles: true,
              cancelable: false,
            })
          )
        }
      )
    }
  }

  handleFieldChange(e) {
    const { onChange } = this.props

    this.setState({
      value: e.target.checked,
    })

    if (onChange) {
      onChange(e)
    }
  }

  handleFieldBlur() {
    this.setState({
      focus: false,
    })
  }

  handleFieldFocus() {
    this.setState({
      focus: true,
    })
  }

  render() {
    const { id, disabled, name, label, error, className } = this.props
    const classes = classNames({
      [this.classesPrefix]: true,
      'is-disabled': disabled,
      checked: this.state.value,
      focused: this.state.focus,
      [`${className}`]: className,
    })
    const componentProps = omit(this.props, [
      'active',
      'placeholder',
      'autoComplete',
      'onChange',
      'onBlur',
      'onKeyDown',
      'label',
      'error',
      'value',
      'focused',
      'checked',
    ])

    return (
      <label
        htmlFor={id}
        className={classes}
        id={`${this.classesPrefix}-${id}`}
      >
        <input
          type="checkbox"
          name={name}
          id={id}
          className="to-input-checkbox"
          {...componentProps}
          onChange={this.handleFieldChange}
          onKeyDown={this.handleKeyDown}
          onBlur={this.handleFieldBlur}
          onFocus={this.handleFieldFocus}
          disabled={disabled}
          checked={this.state.value}
          ref={this.checkbox}
          aria-label={label}
        />
        {label && <div className="to-checkbox-label">{label}</div>}
        {error && <InputError>{error}</InputError>}
      </label>
    )
  }
}

Checkbox.defaultProps = {}

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  value: PropTypes.bool,
  checked: PropTypes.bool,
  focused: PropTypes.bool,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  disabled: PropTypes.bool,
  placeholder: PropTypes.string,
  autofocus: PropTypes.bool,
  readonly: PropTypes.bool,
}

export default Checkbox
