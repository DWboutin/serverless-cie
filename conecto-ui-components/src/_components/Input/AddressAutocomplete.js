import React from 'react'
import PropTypes from 'prop-types'
import omit from 'omit.js'
import nanoidGenerate from 'nanoid/generate'
import classNames from 'classnames'
import PlacesAutocomplete, {
  geocodeByAddress,
  getLatLng,
} from 'react-places-autocomplete'

import Label from './Label'
import InputError from './InputError'
import closeDropdown from '../_helpers/closeDropdown'

export const googleScriptCallbackInit = () => {
  if (typeof window !== 'undefined') {
    window.INIT_GOOGLE_AUTOCOMPLETE = () => {
      if (typeof window.INIT_GOOGLE_AUTOCOMPLETE_FNS !== 'undefined') {
        window.INIT_GOOGLE_AUTOCOMPLETE_FNS.forEach(fn => fn())
      }
    }
  }
}

class AddressAutocomplete extends React.PureComponent {
  constructor(props) {
    super(props)

    this.state = {
      isInitialized: false,
      value: props.value || '',
      focus: false,
    }

    this.input = React.createRef()
    this.autocomplete = React.createRef()

    this.autocompleteUUID = nanoidGenerate(
      '1234567890abcdefghijklmnopqrstuvwxyz',
      10
    )

    this.classesPrefix = 'to-address-autocomplete'

    this.handleSelect = this.handleSelect.bind(this)
    this.handleKeyDown = this.handleKeyDown.bind(this)
    this.handleFieldChange = this.handleFieldChange.bind(this)
    this.handleFieldBlur = this.handleFieldBlur.bind(this)
    this.handleFieldFocus = this.handleFieldFocus.bind(this)
    this.setAdressGeoFromGeo = this.setAdressGeoFromGeo.bind(this)
    this.setAdressStateFromGeo = this.setAdressStateFromGeo.bind(this)
  }

  getSnapshotBeforeUpdate(prevProps, prevState) {
    if (
      this.props &&
      this.props.value &&
      this.props.value.length > 0 &&
      prevProps &&
      prevProps.value !== this.props.value
    ) {
      this.setState({
        value: this.props.value,
      })
    }

    if (this.state.value !== prevState.value) {
      const event = {
        persist: () => {},
        target: {
          type: 'change',
          id: this.props.id,
          name: this.props.name,
          value: this.state.value,
        },
      }

      this.props.onChange(event)
    }

    return null
  }

  componentDidMount() {
    const self = this

    if (typeof window !== 'undefined') {
      if (!window.google) {
        if (typeof window.INIT_GOOGLE_AUTOCOMPLETE_FNS === 'undefined') {
          window.INIT_GOOGLE_AUTOCOMPLETE_FNS = []
        }

        window.INIT_GOOGLE_AUTOCOMPLETE_FNS.push(function() {
          self.setState({
            isInitialized: true,
          })
        })
      } else {
        self.setState({
          isInitialized: true,
        })
      }
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.value !== this.props.value && this.props.value.length === 0) {
      this.setState({
        value: '',
      })
    }
  }

  handleKeyDown(e) {
    const { onPressEnter, onKeyDown } = this.props

    if (e.keyCode === 13 && onPressEnter) {
      onPressEnter(e)
    }

    if (typeof e.keyCode === 'undefined') {
      setTimeout(this.autocomplete.current.fetchPredictions, 50)
    }

    if (onKeyDown) {
      onKeyDown(e)
    }
  }

  handleSelect(address) {
    this.handleFieldChange(address)

    this.setState(
      {
        value: address,
      },
      () => {
        geocodeByAddress(address)
          .then(results => {
            this.setAdressStateFromGeo(results[0])

            return getLatLng(results[0])
          })
          .then(latLng => {
            this.setAdressGeoFromGeo(latLng)
          })
          .catch(error => console.error('Error', error))
      }
    )
  }

  handleFieldChange(value) {
    const { onChange } = this.props

    this.setState({
      value: value,
    })

    if (onChange) {
      onChange(value)
    }
  }

  handleFieldBlur(e) {
    const { onBlur } = this.props

    this.setState({
      value: e.target.value.length > 0 ? e.target.value : '',
      focus: false,
    })

    if (onBlur) {
      onBlur(e)
    }
  }

  handleFieldFocus() {
    this.setState({
      focus: true,
    })
  }

  setAdressStateFromGeo(result) {
    const { onAddressChange } = this.props
    const addressObj = {}

    result.address_components.forEach(addressComponent => {
      if (addressComponent.types.indexOf('street_number') >= 0) {
        addressObj.number = addressComponent.long_name
      } else if (addressComponent.types.indexOf('route') >= 0) {
        addressObj.address = addressComponent.long_name
      } else if (addressComponent.types.indexOf('locality') >= 0) {
        addressObj.city = addressComponent.long_name
      } else if (
        addressComponent.types.indexOf('administrative_area_level_1') >= 0
      ) {
        addressObj.state = addressComponent.short_name
      } else if (addressComponent.types.indexOf('country') >= 0) {
        addressObj.country = addressComponent.short_name
      } else if (addressComponent.types.indexOf('postal_code') >= 0) {
        addressObj.postalCode = addressComponent.long_name
      }
    })

    this.setState({
      correctAddress: addressObj,
      address: result.formatted_address,
    })

    if (onAddressChange) {
      onAddressChange(addressObj)
    }
  }

  setAdressGeoFromGeo(latLng) {
    const { onLatLngChange } = this.props

    this.setState({
      geo: latLng,
    })

    if (onLatLngChange) {
      onLatLngChange(latLng)
    }
  }

  render() {
    const {
      id,
      name,
      label,
      error,
      type,
      className,
      active,
      debounce,
    } = this.props
    const classes = classNames({
      [this.classesPrefix]: true,
      active: this.state.value !== '' || active,
      focused: this.state.focus,
      [`${className}`]: className,
    })
    const componentProps = omit(this.props, [
      'active',
      'label',
      'error',
      'onAddressChange',
      'onLatLngChange',
    ])

    return (
      <div className={classes} id={`${this.classesPrefix}-${id}`}>
        {this.state.isInitialized && (
          <PlacesAutocomplete
            value={this.state.value}
            onChange={this.handleFieldChange}
            onSelect={this.handleSelect}
            onBlur={this.handleFieldBlur}
            debounce={debounce}
            searchOptions={{
              componentRestrictions: {
                country: 'ca',
              },
            }}
            shouldFetchSuggestions={this.state.value.length >= 8}
            ref={this.autocomplete}
          >
            {({
              getInputProps,
              suggestions,
              getSuggestionItemProps,
              loading,
            }) => (
              <>
                <input
                  type={type}
                  name={name}
                  id={id}
                  className="to-input-field"
                  {...componentProps}
                  {...getInputProps()}
                  onFocus={this.handleFieldFocus}
                  onKeyDown={this.handleKeyDown}
                  autoComplete={`autocomplete_off_${this.autocompleteUUID}`}
                  ref={this.input}
                  aria-label={label}
                  aria-labelledby={`${this.classesPrefix}-${id}-label`}
                />
                {label && (
                  <Label
                    htmlFor={id}
                    className="to-input-label"
                    id={`${this.classesPrefix}-${id}-label`}
                  >
                    {label}
                  </Label>
                )}
                <div
                  className={`to-address-autocomplete-choices${
                    suggestions.length > 0 ? ' active' : ''
                  }`}
                >
                  <ul>
                    {loading && <li>Loading...</li>}
                    {suggestions.map(suggestion => {
                      const className = suggestion.active ? 'selected' : ''
                      return (
                        <li
                          {...getSuggestionItemProps(suggestion, {
                            className,
                          })}
                        >
                          {suggestion.description}
                        </li>
                      )
                    })}
                  </ul>
                </div>
              </>
            )}
          </PlacesAutocomplete>
        )}
        {error && <InputError>{error}</InputError>}
      </div>
    )
  }
}

AddressAutocomplete.defaultProps = {
  type: 'text',
  debounce: 450,
}

AddressAutocomplete.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  type: PropTypes.oneOf(['text']),
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  onAddressChange: PropTypes.func,
  onLatLngChange: PropTypes.func,
  disabled: PropTypes.bool,
  placeholder: PropTypes.string,
  autoComplete: PropTypes.string,
  autofocus: PropTypes.bool,
  readonly: PropTypes.bool,
  debounce: PropTypes.number,
}

export default AddressAutocomplete
