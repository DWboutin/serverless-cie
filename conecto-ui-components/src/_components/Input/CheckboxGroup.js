import React from 'react'
import PropTypes from 'prop-types'

import Label from './Label'
import InputError from './InputError'

const CheckboxGroup = ({ label, children, error }) => (
  <div className="to-checkbox-group">
    {label && <Label className="to-checkbox-group-label">{label}</Label>}
    {children}
    {error && <InputError>{error}</InputError>}
  </div>
)

CheckboxGroup.defaultProps = {}

CheckboxGroup.propTypes = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
}

export default CheckboxGroup
