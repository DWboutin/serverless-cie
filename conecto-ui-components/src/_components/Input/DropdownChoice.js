import React from 'react'
import PropTypes from 'prop-types'
import omit from 'omit.js'
import classNames from 'classnames'

import Icon from '../Icon'
import Label from './Label'
import InputError from './InputError'

const DropdownChoice = ({ value, className, selected, children, onClick }) => {
  const classes = classNames({
    'to-dropdown-choice': true,
    selected: selected,
    [`${className}`]: className,
  })

  return (
    <span className={classes} data-value={value} onClick={onClick}>
      {children}
    </span>
  )
}

DropdownChoice.defaultProps = {}

DropdownChoice.propTypes = {
  children: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  selected: PropTypes.bool,
}

export default DropdownChoice
