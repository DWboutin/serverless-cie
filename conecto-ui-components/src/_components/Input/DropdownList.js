import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import Icon from '../Icon'
import Label from './Label'
import InputError from './InputError'

import closeDropdown from '../_helpers/closeDropdown'

class DropdownList extends React.PureComponent {
  constructor(props) {
    super(props)

    this.state = {
      active: false,
      value: props.value || '',
      valueComp: null,
      selectedIndex: 0,
      focus: props.focused || false,
    }

    this.select = React.createRef()
    this.choiceContainer = React.createRef()
    this.choices = []

    this.classesPrefix = 'to-dropdown-list'

    this.addChoiceRef = element => {
      this.choices.push(element)
    }

    this.handleDropdownActivation = this.handleDropdownActivation.bind(this)
    this.handleDropdownClose = this.handleDropdownClose.bind(this)
    this.handleValueSelection = this.handleValueSelection.bind(this)
    this.handleKeyDown = this.handleKeyDown.bind(this)
    this.handleFieldChange = this.handleFieldChange.bind(this)
    this.handleFieldBlur = this.handleFieldBlur.bind(this)
    this.handleFieldFocus = this.handleFieldFocus.bind(this)
  }

  componentDidMount() {
    const { id, value, children } = this.props
    let selectedValue = null

    children.forEach(child => {
      if (child.props.selected === true || child.props.value === value) {
        selectedValue = child.props.value
      }
    })

    if (selectedValue !== null) {
      this.handleValueSelection(selectedValue)
    }

    if (typeof window.TO_DROPDOWN_LISTENER === 'undefined') {
      window.TO_DROPDOWN_LISTENER = true
      window.TO_DROPDOWN_IDS = {
        [`to-dropdown-${id}`]: this.handleDropdownClose,
      }

      document.addEventListener('click', closeDropdown, false)
    } else {
      window.TO_DROPDOWN_IDS[`to-dropdown-${id}`] = this.handleDropdownClose
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.value !== this.props.value) {
      this.handleValueSelection(this.props.value)
    }
  }

  componentWillUnmount() {
    const { id } = this.props

    delete window.TO_DROPDOWN_IDS[`to-dropdown-${id}`]

    if (Object.keys(window.TO_DROPDOWN_IDS).length === 0) {
      delete window.TO_DROPDOWN_LISTENER
      document.removeEventListener('click', closeDropdown, false)
    }
  }

  handleKeyDown(e) {
    // e.keyCode 38 = up | e.keyCode 40 = down
    if ([13, 38, 40].indexOf(e.keyCode) === -1) {
      return false
    } else {
      e.preventDefault()
      const { active, selectedIndex } = this.state

      if (e.keyCode === 13) {
        if (!active) {
          this.setState({
            active: !active,
          })
        } else {
          this.select.current.selectedIndex = selectedIndex
          this.handleValueSelection(
            this.props.children[selectedIndex].props.value
          )
        }
      }

      if (e.keyCode === 38 && selectedIndex > 0) {
        this.setState(
          {
            selectedIndex: selectedIndex - 1,
          },
          () => {
            this.choiceContainer.current.scrollTop =
              this.choices[this.state.selectedIndex].offsetTop +
              40 -
              this.choiceContainer.current.clientHeight
          }
        )
      }

      if (e.keyCode === 40 && selectedIndex < this.props.children.length - 1) {
        this.setState(
          {
            selectedIndex: selectedIndex + 1,
          },
          () => {
            this.choiceContainer.current.scrollTop =
              this.choices[this.state.selectedIndex].offsetTop +
              40 -
              this.choiceContainer.current.clientHeight
          }
        )
      }
    }
  }

  handleDropdownActivation() {
    this.setState({
      active: !this.state.active,
    })
  }

  handleDropdownClose() {
    this.select.current.blur()

    this.setState({
      active: false,
    })
  }

  handleValueSelection(value) {
    let selectedIndex = 0
    const valueComp = this.props.children.filter((child, index) => {
      if (child.props.value === value) {
        selectedIndex = index
        return true
      }

      return false
    })[0]

    this.setState(
      {
        active: false,
        value: value,
        valueComp: valueComp,
        selectedIndex,
      },
      () => {
        // trigger change in select tag
        this.select.current.value = value
        this.select.current.dispatchEvent(
          new Event('change', {
            view: window,
            bubbles: true,
            cancelable: false,
          })
        )
      }
    )
  }

  handleFieldChange(e) {
    const { onChange } = this.props

    this.setState({
      value: e.target.value,
    })

    if (onChange) {
      onChange(e)
    }
  }

  handleFieldBlur(e) {
    const { onBlur } = this.props

    this.setState({
      active: false,
      focus: false,
    })

    if (onBlur) {
      onBlur(e)
    }
  }

  handleFieldFocus() {
    this.setState({
      focus: true,
    })
  }

  render() {
    const { id, name, label, className, error } = this.props
    const classes = classNames({
      [this.classesPrefix]: true,
      active: this.state.active,
      selected: this.state.value.length > 0,
      focused: this.state.focus && !this.state.active,
      [`${className}`]: className,
    })

    return (
      <div className={classes} id={`to-dropdown-${id}`}>
        <div
          className="to-dropdown-list-wrap"
          onClick={this.handleDropdownActivation}
        >
          <div className="text">{this.state.valueComp}</div>
          <select
            id={id}
            name={name}
            ref={this.select}
            value={this.state.value}
            onKeyDown={this.handleKeyDown}
            onChange={this.handleFieldChange}
            onFocus={this.handleFieldFocus}
            onBlur={this.handleFieldBlur}
            aria-label={label}
            aria-labelledby={`${this.classesPrefix}-${id}-label`}
          >
            {React.Children.map(this.props.children, child => {
              return (
                <option value={child.props.value}>
                  {child.props.children}
                </option>
              )
            })}
          </select>
          <div className="arrow">
            <Icon type="arrow-down" />
          </div>
        </div>
        <div className="to-dropdown-list-choices" ref={this.choiceContainer}>
          <ul>
            {React.Children.map(this.props.children, (child, index) => (
              <li
                className={this.state.selectedIndex === index ? 'selected' : ''}
                onClick={() => this.handleValueSelection(child.props.value)}
                ref={this.addChoiceRef}
              >
                {child}
              </li>
            ))}
          </ul>
        </div>
        {label && (
          <Label
            htmlFor={id}
            className="to-dropdown-list-label"
            id={`${this.classesPrefix}-${id}-label`}
          >
            {label}
          </Label>
        )}
        {error && <InputError>{error}</InputError>}
      </div>
    )
  }
}

DropdownList.defaultProps = {
  value: '',
}

DropdownList.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  value: PropTypes.any,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  disabled: PropTypes.bool,
  autofocus: PropTypes.bool,
  readonly: PropTypes.bool,
}

export default DropdownList
