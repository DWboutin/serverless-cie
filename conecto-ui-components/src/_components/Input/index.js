import Input from './Input'
import TextArea from './TextArea'
import Label from './Label'
import InputError from './InputError'
import Checkbox from './Checkbox'
import CheckboxGroup from './CheckboxGroup'
import Switch from './Switch'
import DropdownList from './DropdownList'
import DropdownChoice from './DropdownChoice'
import AddressAutocomplete, {
  googleScriptCallbackInit,
} from './AddressAutocomplete'

export {
  Input as default,
  TextArea,
  Label,
  InputError,
  Checkbox,
  CheckboxGroup,
  Switch,
  DropdownList,
  DropdownChoice,
  AddressAutocomplete,
  googleScriptCallbackInit,
}
