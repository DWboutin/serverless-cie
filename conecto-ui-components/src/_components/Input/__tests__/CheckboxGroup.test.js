import React from 'react'
import { shallow } from 'enzyme'

import { Checkbox, CheckboxGroup } from '../../Input'

global.console = {
  error: jest.fn(),
}

describe('CheckboxGroup', () => {
  beforeEach(() => {
    global.console.error.mockClear()
  })

  it('should match snapshot', () => {
    const wrapper = shallow(
      <CheckboxGroup label="Checkbox group" error="Group error">
        <Checkbox id="testField" name="testField" label="Test field" checked />
        <Checkbox id="testField2" name="testField2" label="Test field 2" />
      </CheckboxGroup>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should console.error when no label is set', () => {
    shallow(
      <CheckboxGroup error="Group error">
        <Checkbox id="testField" name="testField" label="Test field" checked />
        <Checkbox id="testField2" name="testField2" label="Test field 2" />
      </CheckboxGroup>
    )

    expect(global.console.error).toBeCalled()
  })

  it('should console.error when no children are set', () => {
    shallow(<CheckboxGroup label="Checkbox group" error="Group error" />)

    expect(global.console.error).toBeCalled()
  })
})
