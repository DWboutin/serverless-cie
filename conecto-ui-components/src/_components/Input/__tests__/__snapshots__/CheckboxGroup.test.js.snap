// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup should match snapshot 1`] = `
<div
  className="to-checkbox-group"
>
  <Label
    className="to-checkbox-group-label"
  >
    Checkbox group
  </Label>
  <Checkbox
    checked={true}
    id="testField"
    label="Test field"
    name="testField"
  />
  <Checkbox
    id="testField2"
    label="Test field 2"
    name="testField2"
  />
  <InputError>
    Group error
  </InputError>
</div>
`;
